stdlib/sys.pyi

```diff
@@ -74,49 +74,105 @@ if sys.platform == "win32":
     winver: str
 _xoptions: dict[Any, Any]
 
+class _uninstantiable_structseq(structseq[Any]):
+    # type ignore because mypy doesn't like __new__ returning NoReturn
+    def __new__(cls, *args: Any, **kwargs: Any) -> NoReturn: ...  # type: ignore[misc]
```

Comment:
Well, it doesn't look like annotating `__new__` as returning `NoReturn` actually _does_ anything for mypy — mypy raises no errors if you try to instantiate a class annotated as such, and doesn't warn about unreachable code either: https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-codes%2Cstrict&gist=2056f600a3a044d52b769ce816b04708.

A solution that's a little bit of a hack, but works better, would be to annotate these classes as inheriting from `structseq[NoReturn]`.

Or I could stop fussing about it, and just have them inherit from `structseq[Any]`.

Reply:
I don't think this is important either way, because instantiating one of these classes is difficult and unusual anyway. Maybe just use `structseq` everywhere for now?

Reply:
Sure :)

Reply:
What do you think of https://github.com/python/typeshed/pull/6816/commits/7897583bf4f9d226390b95fce6f27f69f0eabeaf?